libs/designer-ui/src/lib/floatingactionmenu/floatingactionmenuinputs/index.tsx

```diff
@@ -137,17 +277,27 @@ export const FloatingActionMenuInputs = (props: FloatingActionMenuInputsProps):
       item.type
     );
     const properties = createDynamicallyAddedParameterProperties(floatingActionMenuItemType, schemaKey);
-    dynamicParameterModels.push({
+
+    const newModel: DynamicallyAddedParameterInputsModel = {
       icon,
       title: properties.title,
       schemaKey,
       properties,
       required: true,
       onRequiredToggle: onDynamicallyAddedParameterRequiredToggle,
+      onStringDropdownListToggle: properties.type === 'string' ? onStringDropdownListToggle : undefined,
+      onStringMultiSelectListToggle: ['array', 'string'].includes(properties.type) ? onStringMultiSelectListToggle : undefined,
       onTitleChange: onDynamicallyAddedParameterTitleChange,
       onDelete: onDynamicallyAddedParameterDelete,
       onRenderValueField,
-    });
+      shouldDisplayAddDropdownOption: properties['x-ms-content-hint'] === 'TEXT',
```

Comment:
This `properties['x-ms-content-hint'] === 'TEXT'` check is performed a LOT of times in this file. Can we assign it to a component-scoped variable?